drivers/arm/pl011/aarch64/pl011_console.S

```diff
@@ -104,41 +164,46 @@ func console_core_putc
 putc_error:
 	mov	w0, #-1
 	ret
-endfunc console_core_putc
+endfunc console_pl011_putc
 
 	/* ---------------------------------------------
-	 * int console_core_getc(uintptr_t base_addr)
+	 * int console_pl011_getc(console_pl011_t *console)
 	 * Function to get a character from the console.
 	 * It returns the character grabbed on success
-	 * or -1 on error.
-	 * In : x0 - console base address
+	 * or -1 if no character is available.
+	 * In : x0 - pointer to console_t struct
+	 * Out: w0 - character if available, else -1
 	 * Clobber list : x0, x1
 	 * ---------------------------------------------
 	 */
-func console_core_getc
-	cbz	x0, getc_error
-1:
+func console_pl011_getc
+#if MULTI_CONSOLE_API
+	ldr	x0, [x0, #CONSOLE_T_PL011_BASE]
+#endif
 	/* Check if the receive FIFO is empty */
 	ldr	w1, [x0, #UARTFR]
-	tbnz	w1, #PL011_UARTFR_RXFE_BIT, 1b
+	tbnz	w1, #PL011_UARTFR_RXFE_BIT, no_char
```

Comment:
minor: relative branches are preferred. So `1f` here.

Reply:
Okay. I wasn't quite clear on which style you prefer... from existing code, I got the impression that you mostly use named labels (except when it's part of a macro).

I feel that named labels can help add some clarity to the code without needing even more explicit comments to describe what a particular check or code section is for everywhere. Doesn't matter much here, but it's more important in more complicated functions like the ones in multi_console.S. Do you want me to change them all to relative labels?

Reply:
I don't feel strongly which style to use in functions; it just seemed strange to change from using a relative label. It's more important to use relative labels in macros, where it's easy to get label collisions.

So it's optional whether you fix this or not.